### DISCUSSION TITLE
Page Renders Twice / Refreshes Initially, only locally

### DISCUSSION THREAD

**Discussion by UmizDemud** [vercel/next.js#56462] [Help]
### Summary (Updating Live)

The landing page I am working on is rendering twice. No solution found yet. The behaviour happens on Chrome and Edge (prob unrelated). I put some time into it with no luck. Result of my test is that it is the next-themes usage. Whn the theme toggler renders, the page refreshes. I will share my repo for inspection. It is a very simple application of a few pages.

The "flickering" is clearly noticable with "Slow 3G" setting, you will also see it loads resources twice over. 

Link to the repo: https://github.com/UmizDemud/Dare-Land-v_23 

Regards

app/layout.tsx
```tsx
  // ...
  return (
    <html lang="en" suppressHydrationWarning>
      <body className={cn(
        "min-h-[100vh] flex flex-col items-stretch",
        inter.className
      )}>
        <ThemeProvider
          attribute="class"
          enableSystem
          defaultTheme="light"
          disableTransitionOnChange
        >
          {children}
        </ThemeProvider>  
      </body>
    </html>
  )
```

Also the navbar.tsx with the theme-toggler:
```tsx
"use client"

import { useEffect, useRef, useState } from 'react'
import useOutsideClick from '@/hooks/useOutsideClick';
import { cn } from '@/lib/utils';
import { MenuButton } from '@/components/ui/menu-button';
import { ContentType, SupportedLanguages } from '@/components/header';
import { ThemeToggle } from "@/components/theme-toggle"
import LanguageDialog from "@/components/ui/language-dialog";
import Link from './nav-link'
import Fallback from './fallback';

interface NavbarProps {
  language: SupportedLanguages
  navbarHeight: string
  cntnt: ContentType[SupportedLanguages]
}

const Navbar: React.FC<NavbarProps> = ({
  language,
  navbarHeight,
  cntnt
}) => {
  const [isMounted, setIsMounted] = useState(false);
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const navMenu = useRef<HTMLDivElement | null>(null);
  const menuButton = useRef<HTMLButtonElement | null>(null);

  const handleMenuClick = () => {
    setIsMenuOpen(prev => !prev)
  }

  useEffect(() => {
    setIsMounted(true);
  }, []);

  useOutsideClick([navMenu, menuButton], () => {
    if (isMenuOpen) {
      menuButton.current?.click();
    }
  });

  if (!isMounted) {
    return (
      <Fallback cntnt={cntnt} navbarHeight={navbarHeight} />
    );
  }

  return (
    <>
      <MenuButton ref={menuButton} isMenuOpen={isMenuOpen} onClick={handleMenuClick} className="md:hidden" />
      <nav
        ref={navMenu}
        className={cn(
          isMenuOpen ? "translate-x-[0]" : "translate-x-[-100%] md:translate-x-0",
          `absolute flex w-3/4 left-0 min-h-[66vh] flex-col gap-4 px-3 border-slate-300 bg-background py-6 transition`,
          "md:w-fit md:min-h-[inherit] md:static md:flex-row md:bg-inherit md:py-0",
          "lg:px-3"
        )}
        style={{zIndex: '999999999', top: navbarHeight}}
      >
        {Object.keys(cntnt.navbar).map((key) => (
          <Link onClick={handleMenuClick} key={key} href={`/${language}/${key}`}>
            {cntnt.navbar[key as keyof typeof cntnt.navbar]}
          </Link>
        ))}
      </nav>
      <div className="md:w-1/5 flex justify-end gap-2">
        <ThemeToggle />
        <LanguageDialog />
      </div>
    </>
  )
}

export default Navbar;
```

### Example
It is deployed on https://dare-land-v-23-7v09y9b9j-umizdemud.vercel.app

**Comment by icyJoseph** [comment]
You can just try to remove the next-themes provider to see if that fixes the issue. If its just a small app. 

I seem to recall someone having this very same issue though, but I cannot find the discussion. I guess what's happening is that next-themes is that it does some client side check to apply the proper theme.

Take a look here: https://github.com/pacocoursey/next-themes/issues/152

**Reply by UmizDemud** [reply]
Yes clearly removing the theme provider stops the refresh, just tested.  But I am trying to use it. I read the thread also, and I'm not seeing an SSR solution with just adding the provider. 

I just shared my provider code above

**Comment by rcteenx** [comment]
favicon file can be a reason.
Chrome request that file.

**Comment by UmizDemud** [comment]
After a few deploys, I do not see the same issue on the deployed site. Really do not know what happened, which means it may have been a cache behaviour. I will rejuvinate the post if I run into more information.
